mm2src/lp_ordermatch.rs

```diff
@@ -3549,6 +3578,110 @@ pub async fn order_status(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, St
         .map_err(|e| ERRL!("{}", e))
 }
 
+enum MakerOrderCancellationReason {
+    Fulfilled,
+    InsufficientBalance,
+    Cancelled,
+}
+enum TakerOrderCancellationReason {
+    Fulfilled,
+    ToMaker,
+    TimedOut,
+    Cancelled,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct MyOrdersFilter {
+    pub order_type: Option<String>,
+    pub initial_action: Option<String>,
+    pub base: Option<String>,
+    pub rel: Option<String>,
+    pub from_price: Option<MmNumber>,
+    pub to_price: Option<MmNumber>,
+    pub from_volume: Option<MmNumber>,
+    pub to_volume: Option<MmNumber>,
+    pub from_timestamp: Option<u64>,
+    pub to_timestamp: Option<u64>,
+    pub was_taker: Option<bool>,
+    pub status: Option<String>,
+    #[serde(default)]
+    pub include_details: bool,
+}
+
+#[derive(Deserialize, Serialize)]
+#[serde(tag = "type", content = "order")]
+enum Order {
+    Maker(MakerOrder),
+    Taker(TakerOrder),
+}
+
+impl<'a> From<&'a Order> for OrderForRpc<'a> {
+    fn from(order: &'a Order) -> OrderForRpc {
+        match order {
+            Order::Maker(o) => OrderForRpc::Maker(MakerOrderForRpc::from(o)),
+            Order::Taker(o) => OrderForRpc::Taker(TakerOrderForRpc::from(o)),
+        }
+    }
+}
+
+#[cfg(target_arch = "wasm32")]
+pub async fn orders_history_by_filter(_ctx: MmArc, _req: Json) -> Result<Response<Vec<u8>>, String> {
+    let res = json!({
+        "error": format!("'orders_history_by_filter' is only supported in native mode yet"),
+    });
+    Response::builder()
+        .status(404)
+        .body(json::to_vec(&res).expect("Serialization failed"))
+        .map_err(|e| ERRL!("{}", e))
+}
+
+/// Returns *all* uuids of swaps, which match the selected filter.
+#[cfg(not(target_arch = "wasm32"))]
+pub async fn orders_history_by_filter(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, String> {
+    use crate::mm2::database::my_orders::select_orders_by_filter;
+
+    let filter: MyOrdersFilter = try_s!(json::from_value(req));
+    let db_result = try_s!(select_orders_by_filter(&ctx.sqlite_connection(), &filter, None));
+
+    let details = if filter.include_details {
+        let mut vec = vec![];
```

Comment:
```rust
Vec::with_capacity(db_result.result.len());
```
Will be more optimal since the required memory will be allocated only once instead of additional allocation per every `.push`.